test/python/WMCore_t/MicroService_t/MSUnmerged_t/MSUnmerged_t.py

```diff
@@ -157,23 +159,46 @@ def testPlineUnmerged(self):
         self.msUnmerged.rseConsStats = self.msUnmerged.rucioConMon.getRSEStats()
         self.msUnmerged.protectedLFNs = set(self.msUnmerged.wmstatsSvc.getProtectedLFNs())
         # Emulate the pipeline run while skipping the last purgeRseObj step
-        self.msUnmerged.resetCounters(plineName=pName)
+        self.msUnmerged.resetServiceCounters()
         rse = self.msUnmerged.updateRSETimestamps(rse, start=True, end=False)
         rse = self.msUnmerged.consRecordAge(rse)
         rse = self.msUnmerged.getUnmergedFiles(rse)
         rse = self.msUnmerged.filterUnmergedFiles(rse)
         rse = self.msUnmerged.cleanRSE(rse)
-        rse = self.msUnmerged.updateRSECounters(rse, pName)
+        rse = self.msUnmerged.updateServiceCounters(rse)
         rse = self.msUnmerged.updateRSETimestamps(rse, start=False, end=True)
         # self.msUnmerged.plineUnmerged.run(rse)
-        expectedRSE = {'counters': {'deletedFail': 0,
-                                    'deletedSuccess': 0,
-                                    'dirsToDelete': 6,
-                                    'dirsToDeleteAll': 6,
-                                    'filesToDelete': 21811,
-                                    'totalNumFiles': 11938},
+        expectedRSE = {'name': 'T2_US_Wisconsin',
+                       'pfnPrefix': None,
+                       'isClean': False,
+                       'rucioConMonStatus': None,
+                       'timestamps': {'endTime': mock.ANY,
+                                      'prevEndTime': 0.0,
+                                      'prevStartTime': 0.0,
+                                      'rseConsStatTime': mock.ANY,
+                                      'startTime': mock.ANY},
+                       "counters": {"totalNumFiles": 11938,
+                                    "totalNumDirs": 11,
+                                    "dirsToDelete": 6,
+                                    "filesToDelete": 0,
+                                    "filesDeletedSuccess": 0,
+                                    "filesDeletedFail": 0,
+                                    "dirsDeletedSuccess": 0,
+                                    "dirsDeletedFail": 0,
+                                    "gfalErrors": {}},
+                       'files': {'allUnmerged': mock.ANY,
+                                 'deletedFail': set(),
+                                 'deletedSuccess': set(),
+                                 'protected': {},
+                                 'toDelete': {'/store/unmerged/Phase2HLTTDRSummer20ReRECOMiniAOD/DYToLL_M-50_TuneCP5_14TeV-pythia8/FEVT/FlatPU0To200_pilot_111X_mcRun4_realistic_T15_v1-v2': mock.ANY,
```

Comment:
Hi @vkuznet So there is argumentation behind the document schema (which is here BTW: https://github.com/dmwm/WMCore/blob/14e381381dd4639a26fb61807c2f04cd93cf9807/src/python/WMCore/MicroService/MSUnmerged/MSUnmergedRSE.py#L40-L69) . The dynamic keys are needed as a place holder for the generators of filters to the final list of files to be deleted. We have extremely long lists per RSE and duplicating the `toDelete` lists inside the object was only leading to extreme memory footprint. Instead we preserve the full list only at one place and we dynamically create filters at runtime based on the current conditions (what is and what is not protected), then we keep only the filters. You see it 'hardcoded' here as such only because this is a unit test with predefined data for which I have tested in advance what should be the name to be expect here. In the real code those filters are generated at runtime, hence the dynamic names of the keys to hold them.

Reply:
Please show me concrete example. And, justify why do you need to keep LFNs per RSE in single document? Why not to use separate document for that and then refer (a la foreign key in RDBMS world) to it. What are those generators and filters? And, which duplication you're talking about. Sorry, so far nothing is clear to me how you convert your static schema to dynamic one, by avoiding some duplication via dynamic keys in a dictionary.

Reply:
And, why do you keep schema within specific class? Can you define it separately and load it in a class? Here we are having similar situation to `Lexicon.py` issue, which now we're trying to share among different code bases and it is extremely difficult since of embedded nature of the schema. The schema files defines the data. As such they should not be part of concrete implementation and it is better to keep it separately to allow to have different implementation for that particular schema.

Reply:
Given that this comment refers to a test module, and that it goes beyond the scope of this PR, I would suggest to follow this up offline. Or in one of our future reviews of the system, services, etc. What do you all think?

Reply:
Completely agree with you here @amaltaro